static/main.js

```diff
@@ -85,15 +85,80 @@ function init() {
         }
     });
 
+    // This sets default values
+    // Initialize an empty array to store the options as strings
+    const optionElements = [];
+
+    // Init array with ids of each option
+    const optionIds = ["displayMode", "leqno", "fleqn", "throwOnError",
+        "errorColor", "strict", "output", "trust", "macros"];
+
+    // Loop through each option ID
+    for (const id of optionIds) {
+
+        const element = document.getElementById(id);
+        // If value of element changes, call reprocess function
+        element.addEventListener("change", reprocess);
+        if (element.type === "checkbox") {
+            // if checkbox set its checked state based on options[id]
+            element.checked = options[id];
+        } else if (options[id]) {
+            // If id property of options is defined
+            if (element.placeholder === "JSON") {
+                // check if placeholder is "JSON"
+                // this means it is macros
+                // convert Js object to JSON string
+                element.value = JSON.stringify(options[id]);
+            } else {
+                //if not defined
+                element.value = options[id];
+            }
+        }
+
+        // Add the processed element to the optionElements array
+        optionElements.push(element);
+    }
+
+
     reprocess();
 
+    function changeOption() {
+        // Init empty object options
+        const options = {};
```

Comment:
I don't see why you need to make a new `options` object, instead of modifying the original one. I think the latter approach would be more robust, in case `options` has some hidden options that are not reflected by the UI.